101-function-app-storage-private-endpoints/azuredeploy.json

```diff
@@ -0,0 +1,674 @@
+      "dependsOn": [
+        "[resourceId('Microsoft.Web/sites', parameters('AppName'))]",
+        "[resourceId('Microsoft.Network/virtualNetworks', parameters('vnetName'))]"
+      ],
+      "properties": {
+        "subnetResourceId": "[resourceId('Microsoft.Network/virtualNetworks/subnets', parameters('vnetName'), parameters('functionsSubnetName'))]",
+        "isSwift": true
+      }
+    },
+    {
+      "type": "Microsoft.Web/sites/config",
+      "name": "[concat(parameters('AppName'), '/web')]",
+      "apiVersion": "2020-09-01",
+      "dependsOn": [
+        "[resourceId('Microsoft.Web/sites', parameters('AppName'))]",
+        "[resourceId('Microsoft.Network/virtualNetworks', parameters('vnetName'))]"
+      ],
+      "properties": {
+        "functionsRuntimeScaleMonitoringEnabled": true
+      }
+    },
+    {
+      "type": "Microsoft.Resources/deploymentScripts",
+      "apiVersion": "2020-10-01",
+      "name": "AzPowerShellStartHost",
+      "condition": "[equals(parameters('postDeploymentScript'),'azpowershell')]",
+      "location": "[parameters('Location')]",
+      "kind": "AzurePowerShell",
+      "dependsOn": [
+        "[resourceId('Microsoft.Web/sites/config', parameters('AppName'), 'web')]"
+      ],
+      "identity": {
+        "type": "userAssigned",
+        "userAssignedIdentities": {
+          "[resourceId(parameters('userIdentityResourceGroup'), 'Microsoft.ManagedIdentity/userAssignedIdentities', parameters('userIdentity'))]": {}
+        }
+      },
+      "properties": {
+        "azPowerShellVersion": "3.0",
+        "arguments": "[format(' -appName {0} -planName {1} -planSize {2} -rgName {3}', parameters('AppName'), parameters('plan'), parameters('finalplansize'), resourceGroup().name)]",
+        "scriptContent": "param([string][Parameter(Mandatory=$true)] $appName, [string][Parameter(Mandatory=$true)] $planName, [string][Parameter(Mandatory=$true)] $planSize, [string][Parameter(Mandatory=$true)] $rgName); $SkuObject = @{\"name\" = $planSize}; Set-AzResource -ApiVersion 2020-09-01 -Sku $SkuObject -Name $planName -ResourceGroupName $rgName -ResourceType Microsoft.Web/serverfarms -Force; $publishProfileString = Invoke-AzResourceAction -ResourceGroupName $rgName -ResourceType Microsoft.Web/sites -ResourceName $appName -Action publishxml -ApiVersion 2020-09-01 -Force; $publishProfileXml =[xml]$publishProfileString; $username = $publishProfileXml.publishData.FirstChild.userName; $password = $publishProfileXml.publishData.FirstChild.userPWD; $base64AuthInfo = [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes((\"{0}:{1}\" -f $username,$password))); $userAgent = \"powershell/1.0\"; $apiUrl = \"https://$appName.scm.azurewebsites.net/DebugConsole\"; Invoke-RestMethod -Uri $apiUrl -Headers @{Authorization=(\"Basic {0}\" -f $base64AuthInfo)} -UserAgent $userAgent -Method GET;",
+        "timeout": "PT5M",
+        "cleanupPreference": "OnSuccess",
+        "retentionInterval": "P1D"
+      }
+    },
+    {
```

Comment:
are these scripts doing something different?  if so, why are they mutually exclusive?  If not, why do we need 2?

Reply:
Both scripts do the same thing, but one does the operations via Az PowerShell whereas the other does the same operations via Az CLI. I wanted to offer both options in case the the user's deployment environment has one but not the other. I can make it just Az CLI if needed and take out the bash-specific parts

Reply:
I think either is fine, unedited if they are the same.  It's here to "complete" deployment so if it uses a language that's not familiar (but allowed) that's fine.

Reply:
thanks, just realized I could/should use the 'if' function to rule this up into one script section. Will work on that along with the other changes.